modules/drivers/audio/respeaker.h

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#pragma once
+
+#include <google/protobuf/message.h>
+#include <portaudio.h>
+
+#include <memory>
+#include <string>
+
+#include "cyber/cyber.h"
+#include "modules/drivers/audio/proto/speaker_config.pb.h"
+
+namespace apollo {
+namespace drivers {
+namespace audio {
+
+using apollo::drivers::audio::config::SpeakerConfig;
+
+/* An incomplete version of PA Stream used only for respeaker as input.
+ * Refer to http://portaudio.com/docs/v19-doxydocs/ for more information
+ */
+class Stream {
+ private:
+  PaStream *pastream_ptr_;
+  PaStreamParameters *inputParameters_ptr_;
+
+ public:
+  Stream() {}
+  ~Stream();
+  void init_stream(int rate, int channels, int chunk, int input_device_index,
+                   PaSampleFormat format);
+  void read_stream(int n_frames, char *buffer);
+  int get_chunk_size(int n_frames);
+};
+
+class Respeaker {
+ public:
+  std::unique_ptr<Stream> stream_ptr_;
```

Comment:
Should stream_ptr_ be public ?  Have this header file passed our lint check ( by running `./apollo.sh lint`) ?

Reply:
Yes it did pass `./apollo.sh lint`.
I used it as public because otherwise I have to encapsulate member functions of `Stream` in `Respeaker` with the same name. (like `int Respeaker::get_chunk_size(arg){pastream_->get_chunk_size(arg)};`) Do you think it's fine? Or do you have any suggestions? Thank you.

Reply:
Yes please do so and make the return value/ function as a const

Reply:
> Yes please do so and make the return value/ function as a const

Done. I kind of `const` everything if possible...